### PR TITLE
Listener node

### PR DESCRIPTION
**Pull request by itkg-alavieille**

### CODE REVIEW

File: GroupBundle/EventListener/AbstractNodeGroupRoleListener.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * @param string $nodeId
+     * @param string $role
+     * @param string $accessType
+     *
+     * @return NodeGroupRoleInterface
+     */
+    protected function createNodeGroupRole($nodeId, $role, $accessType)
+    {
+
```
Comment (nicolasThal) [inline GroupBundle/EventListener/AbstractNodeGroupRoleListener.php]: empty line

File: GroupBundle/EventListener/CreateGroupListener.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace OpenOrchestra\GroupBundle\EventListener;
+
+use Doctrine\ODM\MongoDB\Event\LifecycleEventArgs;
+use OpenOrchestra\BackofficeBundle\Model\GroupInterface;
+use OpenOrchestra\ModelInterface\Model\SiteInterface;
+
+/**
+ * Class CreateGroupListener
+ */
+class CreateGroupListener extends AbstractNodeGroupRoleListener
```
Comment (nicolasThal) [inline GroupBundle/EventListener/CreateGroupListener.php]: I think that the name should be related to what the listener does and not to what the listener listen to

File: GroupBundle/EventListener/CreateNodeListener.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+namespace OpenOrchestra\GroupBundle\EventListener;
+
+use Doctrine\ODM\MongoDB\Event\LifecycleEventArgs;
+use OpenOrchestra\BackofficeBundle\Model\GroupInterface;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+
+/**
+ * Class CreateNodeListener
+ */
+class CreateNodeListener extends AbstractNodeGroupRoleListener
```
Comment (nicolasThal) [inline GroupBundle/EventListener/CreateNodeListener.php]: same

File: GroupBundle/EventListener/CreateNodeListener.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                foreach ($nodesRoles as $role) {
+                    $nodeGroupRole = $this->createNodeGroupRole($document->getNodeId(), $role, $accessType);
+                    $group->addNodeRole($nodeGroupRole) ;
+                    $event->getDocumentManager()->persist($group);
+                    $event->getDocumentManager()->flush($group);
+                }
+            }
+        }
+    }
+}
```
Comment (nicolasThal) [inline GroupBundle/EventListener/CreateNodeListener.php]: eof

File: GroupBundle/Tests/EventListener/AbstractNodeGroupRoleListenerTest.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+     */
+    public function setUp()
+    {
+        $this->lifecycleEventArgs = Phake::mock('Doctrine\ODM\MongoDB\Event\LifecycleEventArgs');
+        $this->container = Phake::mock('Symfony\Component\DependencyInjection\Container');
+        $roleCollector = Phake::mock('OpenOrchestra\Backoffice\Collector\RoleCollector');
+        Phake::when($this->container)->get('open_orchestra_backoffice.collector.role')->thenReturn($roleCollector);
+        Phake::when($roleCollector)->getRolesByType(Phake::anyParameters())->thenReturn($this->nodesRoles);
+    }
+
```
Comment (nicolasThal) [inline GroupBundle/Tests/EventListener/AbstractNodeGroupRoleListenerTest.php]: empty line to suppress

File: GroupBundle/EventListener/AbstractNodeGroupRoleListener.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function __construct($nodeGroupRoleClass)
+    {
+        $this->nodeGroupRoleClass = $nodeGroupRoleClass;
+    }
+
+    /**
+     * @return array
+     */
+    public function getNodeRoles()
```
Comment (nicolasThal) [inline GroupBundle/EventListener/AbstractNodeGroupRoleListener.php]: does this method has to be public ?

**Comment by itkg-alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Comment by itkg-nanne** [comment]
:+1:

**Pull request merged**